Adds a featured post on algebraic effects

mdimjasevic opened the pull request:
This PR starts an issue 258 of Haskell Weekly by adding a featured post that motivates algebraic effects in Haskell.

tfausak commented:
Nice, I enjoyed your post! I have two comments about this PR though:

- Would you mind adding a byline to your entry?
- Could you re-add the `undefined` bits to the other sections? I use them to remind myself to fill those sections out :)

mdimjasevic commented:
> Nice, I enjoyed your post! I have two comments about this PR though:
> 
>     * Would you mind adding a byline to your entry?
> 
>     * Could you re-add the `undefined` bits to the other sections? I use them to remind myself to fill those sections out :)

Thank you, Taylor!

I added the byline. I am not sure if `undefined` was literally having that word in each empty section or something else, but I added that word literally. Let me know if I got it wrong.
